source/codegen/metadata_validation.py

```diff
@@ -113,6 +113,14 @@ class RULES:
         Optional("linked_params_are_optional"): bool,
         Optional("mapped-enum"): str,
         Optional("is_optional"): bool,
+        Optional("optional"): bool,
+        Optional("ctypes_data_type"): Or(str, None),
+        Optional("has_explicit_buffer_size"): bool,
+        Optional("is_list"): bool,
```

Comment:
FYI, there is `use_array` above, but I'm not sure if it means the same thing. None of the metadata in grpc-device currently uses it.

Reply:
Generally, `is_list` explains whether the parameter is a list or not.
And having key name to `is_list` seems to be more apt than `use_array`.

Reply:
I think the distinction is pretty much language-specific: in the C API, it's an array, and in the Python API, it's a list. But I don't expect you to change this.